pallets/qa-tools/src/pallet_tools/liquidity_proxy.rs

```diff
@@ -0,0 +1,56 @@
+pub mod source_initializers {
+    use crate::{settings, Config};
+    use codec::{Decode, Encode};
+    use common::prelude::BalanceUnit;
+    use frame_support::dispatch::DispatchResult;
+    use frame_system::pallet_prelude::BlockNumberFor;
+    use order_book::{MomentOf, OrderBookId};
+    use sp_std::fmt::Debug;
+    use sp_std::vec::Vec;
+
+    #[derive(Clone, PartialEq, Eq, Encode, Decode, scale_info::TypeInfo, Debug)]
+    #[scale_info(skip_type_params(T))]
+    pub struct XYKPair<DEXId, AssetId> {
+        pub dex_id: DEXId,
+        pub asset_a: AssetId,
+        pub asset_b: AssetId,
+        /// Price of `asset_a` in terms of `asset_b` (how much `asset_b` is needed to buy 1 `asset_a`)
+        pub price: BalanceUnit,
```

Comment:
Probably we can use just `Balance` type here, because XYK pool works only with divisible assets.